packages/framework/src/Foundation/Internal/LoadYamlConfiguration.php

```diff
@@ -80,14 +92,26 @@ protected function mergeParsedConfiguration(): void
 
     protected function mergeConfiguration(string $namespace, array $yamlData): void
     {
-        Config::set($namespace, array_merge(
-            Config::getArray($namespace, []),
+        $this->config[$namespace] = array_merge(
+            $this->config[$namespace] ?? [],
             $yamlData
-        ));
+        );
     }
 
     protected function configurationContainsNamespaces(array $yaml): bool
     {
         return array_key_first($yaml) === 'hyde';
     }
+
+    private function supportSettingSiteNameSetsSidebarHeaderOption(): void
+    {
+        $sidebarHeaderIsNotSetInPhpConfig = ($this->config['docs']['sidebar']['header'] ?? null) === 'HydePHP Docs';
+        $siteNameFromYaml = $this->configurationContainsNamespaces($this->yaml) ? ($this->yaml['hyde']['name'] ?? null) : ($this->yaml['name'] ?? null);
```

Comment:
We could also check environment values for the site name, but I think that adds to much complexity.

Reply:
Might be worth to at least try to implement this and see how complex it is, seeing as it would mean the `env-site-name` setting in the https://github.com/hydephp/action would then work for this automatically.

Reply:
Question is the, if it's set in both Yaml and .env, which takes precedence? I think .env as that usually has precedence generally speaking

Reply:
Wait, I was wrong again. This should already be implicitly supported by the very fact of an environment variable being used. Originally I considered if we should do things the other way around but that's why we closed https://github.com/hydephp/develop/pull/1691